transaction/tests/StronglyConnectedComponents_unittest.cpp

```diff
@@ -0,0 +1,234 @@
+                                           12,
+                                           edge_mapping);
+
+    // Run the SCC algorithm.
+    scc_ = std::make_unique<StronglyConnectedComponents>(*wait_for_graph_);
+  }
+
+  std::unique_ptr<GraphConfiguration> graph_configuration_;
+  std::unique_ptr<DirectedGraph> wait_for_graph_;
+  std::unique_ptr<StronglyConnectedComponents> scc_;
+};
+
+TEST_F(StronglyConnectedComponentsTestWithOneNode, TotalNumberOfComponents) {
+  std::uint64_t total_components = scc_->getTotalComponents();
+
+  EXPECT_EQ(1u, total_components);
+}
+
+TEST_F(StronglyConnectedComponentsTestWithOneNode, GetComponentId) {
+  std::uint64_t nid0_component =
+    scc_->getComponentId(graph_configuration_->getNodesList()[0]);
+
+  EXPECT_EQ(0u, nid0_component);
+}
+
+
+TEST_F(StronglyConnectedComponentsTestWithOneNode, GetComponentsMapping) {
+  std::unordered_map<std::uint64_t, std::vector<DirectedGraph::node_id>>
+    mapping = scc_->getComponentMapping();
+  std::vector<DirectedGraph::node_id> component_no_0 = mapping[0];
+
+  EXPECT_EQ(1u, component_no_0.size());
+}
+
+TEST_F(StronglyConnectedComponentsTestWithTwoNodesCycle, TotalNumberOfComponents) {
+  std::uint64_t total_components = scc_->getTotalComponents();
+
+  EXPECT_EQ(1u, total_components);
+}
+
+TEST_F(StronglyConnectedComponentsTestWithTwoNodesCycle, GetComponentId) {
+  DirectedGraph::node_id nid0 = DirectedGraph::node_id(0);
+  DirectedGraph::node_id nid1 = DirectedGraph::node_id(1);
+
+  // Since two nodes make a cycle, they should have same component id.
+  EXPECT_EQ(0u, scc_->getComponentId(nid0));
+  EXPECT_EQ(0u, scc_->getComponentId(nid1));
+}
+
+TEST_F(StronglyConnectedComponentsTestWithTwoNodesCycle, GetComponentsMapping) {
+  std::unordered_map<std::uint64_t, std::vector<DirectedGraph::node_id>>
+      mapping = scc_->getComponentMapping();
+  std::vector<DirectedGraph::node_id> component_no_0 = mapping[0];
+
+  // Since there are 2 nodes, and they are in the same SC component,
+  // the component size is 2.
+  EXPECT_EQ(2u, component_no_0.size());
+}
+
+TEST_F(StronglyConnectedComponentsTest, TotalNumberOfComponents) {
+  std::size_t total_components = scc_->getTotalComponents();
+
+  EXPECT_EQ(4u, total_components);
+}
+
+TEST_F(StronglyConnectedComponentsTest, GetComponentId) {
+  std::vector<DirectedGraph::node_id> nid_list;
+  for (DirectedGraph::node_id nid = 0;
+       nid < wait_for_graph_->getNumNodes();
+       ++nid) {
+    nid_list.push_back(scc_->getComponentId(nid));
+  }
+
+  // Check the result are as expected.
+  EXPECT_EQ(3u, nid_list[0]);
+
+  EXPECT_EQ(2u, nid_list[1]);
+  EXPECT_EQ(2u, nid_list[3]);
+  EXPECT_EQ(2u, nid_list[4]);
+
+  EXPECT_EQ(1u, nid_list[2]);
+  EXPECT_EQ(1u, nid_list[5]);
+
+  EXPECT_EQ(0u, nid_list[6]);
+  EXPECT_EQ(0u, nid_list[7]);
+  EXPECT_EQ(0u, nid_list[8]);
+  EXPECT_EQ(0u, nid_list[9]);
+  EXPECT_EQ(0u, nid_list[10]);
+  EXPECT_EQ(0u, nid_list[11]);
```

Comment:
You need not do it, but the above set of ``EXPECT_EQ`` can be made even more compact. You can create a vector called expected_component_count and initialize it with ``{3, 2, 2, ...0}``. Then in a loop, you can check the equality of ``expected_component_count[i]`` and ``nid_list[i]``